cpp/src/barretenberg/honk/pcs/gemini/gemini.hpp

```diff
@@ -100,35 +76,20 @@ template <typename Params> class MultilinearReductionScheme {
 
   public:
     /**
-     * @brief reduces claims about multiple (shifted) MLE evaluation
+     * @brief Computes d-1 fold polynomials Fold_i, i = 1, ..., d-1
      *
-     * @param ck is the commitment key for creating the new commitments
      * @param mle_opening_point u = (u₀,...,uₘ₋₁) is the MLE opening point
-     * @param batched_shifted batch polynomial constructed from the unshifted multivariates
-     * @param batched_to_be_shifted batch polynomial constructed from the to-be-shifted multivariates
-     * @param transcript
-     * @return Output (opening pairs, folded_witness_polynomials)
+     * @param fold_polynomials (m + 1) - many polynomials. The first two are assumed initialized to F(X) = ∑ⱼ ρʲfⱼ(X)
+     * and G(X) = ∑ⱼ ρᵏ⁺ʲ gⱼ(X), i.e. the batched unshifted polynomials and the batched to-be-shifted polynomials. This
+     * function populates the next d-1 elements with Fold_i, i = 1, ..., d-1.
      *
      */
-    static ProverOutput<Params> reduce_prove(std::shared_ptr<CK> ck,
-                                             std::span<const Fr> mle_opening_point,
-                                             const Polynomial&& batched_shifted,       /* unshifted */
-                                             const Polynomial&& batched_to_be_shifted, /* to-be-shifted */
-                                             ProverTranscript<Fr>& transcript)
+    static void compute_fold_polynomials(std::span<const Fr> mle_opening_point, auto& fold_polynomials)
```

Comment:
Small nitpick, but I'm not really a fan of using arguments for return values. We could pass `batched_shifted` and `batched_to_be_shifted` by move, and return the vector of `fold_polynomials`

Reply:
No you're right, that's better. Will make the change

Reply:
Done